airflow/api_connexion/schemas/dag_run_schema.py

```diff
@@ -121,6 +121,18 @@ def autofill(self, data, **kwargs):
 
         return ret_data
 
+    @validates_schema
+    def validate_data_interval_dates(self, data, **kwargs):
+        data_interval_start_exists = data.get("data_interval_start") is not None
+        data_interval_end_exists = data.get("data_interval_end") is not None
+
+        if (data_interval_start_exists and not data_interval_end_exists) or (
+            data_interval_end_exists and not data_interval_start_exists
+        ):
```

Comment:
```suggestion
        if data_interval_start_exists != data_interval_end_exists:
```

Less readable on its own, but the error below should make the intention clear.